certstore/certstore.go

```diff
@@ -143,21 +280,13 @@ func (cs *Store) Put(ctx context.Context, cert *certs.FinalityCertificate) error
 		return xerrors.Errorf("marshalling cert instance %d: %w", cert.GPBFTInstance, err)
 	}
 
-	cs.writeLk.Lock()
-	defer cs.writeLk.Unlock()
-
-	if cs.Latest() != nil && cert.GPBFTInstance > cs.Latest().GPBFTInstance &&
-		cert.GPBFTInstance != cs.Latest().GPBFTInstance+1 {
-		return xerrors.Errorf("attempted to add cert at %d but the previous one is %d",
-			cert.GPBFTInstance, cs.Latest().GPBFTInstance)
-	}
-
-	if err := cs.ds.Put(ctx, key, buf.Bytes()); err != nil {
+	if err := cs.ds.Put(ctx, cs.keyForCert(cert.GPBFTInstance), buf.Bytes()); err != nil {
 		return xerrors.Errorf("putting the cert: %w", err)
 	}
-	if err := cs.ds.Put(ctx, certStoreLatestKey, buf.Bytes()); err != nil {
-		return xerrors.Errorf("putting the cert: %w", err)
+	if err := cs.writePointer(ctx, certStoreLatestKey, cert.GPBFTInstance); err != nil {
```

Comment:
I now write the instance number of the latest finality certificate, instead of writing the actual certificate (more efficient, matches what I'm doing with the "first instance").